sw/device/boot_rom/rom_crt.S

```diff
@@ -75,18 +81,40 @@ _reset_start:
 _start:
   .globl _start
 
+  // Enable entropy complex - this is not the full enable
+  li   a0, TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR
+  li   t0, 0x2
+  sw   t0, ENTROPY_SRC_CONF_REG_OFFSET(a0)
+
+  li   a0, TOP_EARLGREY_CSRNG_BASE_ADDR
+  li   t0, 0x1
+  sw   t0, CSRNG_CTRL_REG_OFFSET(a0)
+
+  li   a0, TOP_EARLGREY_EDN0_BASE_ADDR
+  li   t0, 0x1
+  sw   t0, EDN_CTRL_REG_OFFSET(a0)
+
+  // Request memory scrambling and init
+  li   a0, TOP_EARLGREY_SRAM_CTRL_MAIN_BASE_ADDR
+  li   t0, 1
+  slli t1, t0, SRAM_CTRL_CTRL_RENEW_SCR_KEY_BIT
```

Comment:
I think you can simplify this since these are constants:

```asm
li a0, TOP_EARLGREY_SRAM_CTRL_MAIN_BASE_ADDR
li a1, (1<<SRAM_CTRL_CTRL_RENEW_SCR_KEY_BIT)|(1<<SRAM_CTRL_CTRL_INIT_BIT)
sw   a1, SRAM_CTRL_CTRL_REG_OFFSET(a0)
```
```

Reply:
thanks! simplified.